### PR TITLE
En 10372 elrond go core integration

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
- integrate elrond-go-core into elrond-go
- multiple fixes
- moved trie and state components  + interfaces in separate packages

### CODE REVIEW

File: update/errors.go [hunk 1]
```diff
@@ -50,8 +50,8 @@ var ErrNilHasher = errors.New("nil Hasher")
 // ErrNilShardCoordinator signals that an operation has been attempted to or with a nil shard coordinator
 var ErrNilShardCoordinator = errors.New("nil shard coordinator")
 
-// ErrNilCoreComponents signals that an operation has been attempted with nil core components
-var ErrNilCoreComponents = errors.New("nil core components")
+// ErrNilCoreComponents signals that an operation has been attempted with nil common components
```
Comment (iulianpascalau) [inline update/errors.go:53]: 2 unwanted changes

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: process/txsimulator/wrappedAccountsDB.go [hunk 2]
```diff
@@ -78,11 +78,11 @@ func (w *readOnlyAccountsDB) RecreateTrie(_ []byte) error {
 }
 
 // PruneTrie won't do anything as write operations are disabled on this component
-func (w *readOnlyAccountsDB) PruneTrie(_ []byte, _ data.TriePruningIdentifier) {
+func (w *readOnlyAccountsDB) PruneTrie(_ []byte, _ temporary.TriePruningIdentifier) {
```
Comment (miiu96) [inline process/txsimulator/wrappedAccountsDB.go:81]: rad instead of w

File: common/interface.go [hunk 3]
```diff
@@ -111,3 +103,14 @@ type SafeCloser interface {
 	ChanClose() <-chan struct{}
 	IsInterfaceNil() bool
 }
+
+// Logger defines the behavior of a data logger component
+type Logger interface {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply common/interface.go]: it was not used. removed

File: common/interface.go [hunk 4]
```diff
@@ -71,6 +57,12 @@ type KeyValueHolder interface {
 	ValueWithoutSuffix(suffix []byte) ([]byte, error)
 }
 
+// EpochSubscriberHandler defines the behavior of a component that can be notified if a new epoch was confirmed
+type EpochSubscriberHandler interface {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply common/interface.go]: it was not used. removed

File: consensus/spos/errors.go [hunk 5]
```diff
@@ -29,7 +29,7 @@ var ErrNilMultiSigner = errors.New("multiSigner is nil")
 var ErrNilConsensusState = errors.New("consensus state is nil")
 
 // ErrNilConsensusCore is raised when a valid ConsensusCore is expected but nil used
-var ErrNilConsensusCore = errors.New("consensus core is nil")
+var ErrNilConsensusCore = errors.New("consensus common is nil")
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply consensus/spos/errors.go:32]: reverted

File: epochStart/errors.go [hunk 6]
```diff
@@ -92,8 +92,8 @@ var ErrNilShardCoordinator = errors.New("shard coordinator is nil")
 // ErrNilPubkeyConverter signals that nil address converter was provided
 var ErrNilPubkeyConverter = errors.New("nil pubkey converter")
 
-// ErrNilCoreComponentsHolder signals that a nil core components holder was provided
-var ErrNilCoreComponentsHolder = errors.New("nil core components holder")
+// ErrNilCoreComponentsHolder signals that a nil common components holder was provided
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply epochStart/errors.go:95]: reverted

File: errors/errors.go [hunk 7]
```diff
@@ -89,8 +89,8 @@ var ErrNilConsensusComponentsFactory = errors.New("nil consensus components fact
 // ErrNilCryptoComponentsFactory signals that the provided crypto components factory is nil
 var ErrNilCryptoComponentsFactory = errors.New("nil crypto components factory")
 
-// ErrNilCoreComponentsFactory signals that the provided core components factory is nil
-var ErrNilCoreComponentsFactory = errors.New("nil core components factory")
+// ErrNilCoreComponentsFactory signals that the provided common components factory is nil
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply errors/errors.go:92]: reverted

File: examples/construction_test.go [hunk 8]
```diff
@@ -5,17 +5,18 @@ import (
...
 	"github.com/ElrondNetwork/elrond-go/crypto/signing/ed25519"
 	"github.com/ElrondNetwork/elrond-go/crypto/signing/ed25519/singlesig"
-	"github.com/ElrondNetwork/elrond-go/data/transaction"
-	"github.com/ElrondNetwork/elrond-go/hashing/blake2b"
-	"github.com/ElrondNetwork/elrond-go/marshal"
 	"github.com/stretchr/testify/require"
 )
 
-var addressEncoder, _ = pubkeyConverter.NewBech32PubkeyConverter(32)
+var addressEncoder, _ = pubkeyConverter.NewBech32PubkeyConverter(32, &mock.LoggerFake{})
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply examples/construction_test.go]: done

File: factory/coreComponents.go [hunk 9]
```diff
@@ -5,38 +5,38 @@ import (
...
 	"github.com/ElrondNetwork/elrond-go/process/rating"
 	"github.com/ElrondNetwork/elrond-go/process/smartContract"
 	"github.com/ElrondNetwork/elrond-go/sharding"
+	stateFactory "github.com/ElrondNetwork/elrond-go/state/factory"
 	"github.com/ElrondNetwork/elrond-go/storage"
 	storageFactory "github.com/ElrondNetwork/elrond-go/storage/factory"
 )
 
-// CoreComponentsFactoryArgs holds the arguments needed for creating a core components factory
+// CoreComponentsFactoryArgs holds the arguments needed for creating a common components factory
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply factory/coreComponents.go:39]: reverted

File: factory/coreComponentsHandler.go [hunk 10]
```diff
@@ -24,14 +24,14 @@ var _ ComponentHandler = (*managedCoreComponents)(nil)
 var _ CoreComponentsHolder = (*managedCoreComponents)(nil)
 var _ CoreComponentsHandler = (*managedCoreComponents)(nil)
 
-// managedCoreComponents is an implementation of core components handler that can create, close and access the core components
+// managedCoreComponents is an implementation of common components handler that can create, close and access the common components
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply factory/coreComponentsHandler.go:27]: reverted

File: factory/interface.go [hunk 11]
```diff
@@ -87,7 +88,7 @@ type ComponentHandler interface {
 	CheckSubcomponents() error
 }
 
-// CoreComponentsHolder holds the core components
+// CoreComponentsHolder holds the common components
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply factory/interface.go:91]: reverted

File: genesis/mock/storageManagerStub.go [hunk 12]
```diff
@@ -1,27 +1,27 @@
...
+	MarkForEvictionCalled             func([]byte, temporary.ModifiedHashes) error
+	GetDbThatContainsHashCalled       func([]byte) temporary.DBWriteCacher
+	GetSnapshotThatContainsHashCalled func(rootHash []byte) temporary.SnapshotDbHandler
 	IsPruningEnabledCalled            func() bool
 	EnterSnapshotModeCalled           func()
 	ExitSnapshotModeCalled            func()
 	IsInterfaceNilCalled              func() bool
 }
 
 // Database --
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply genesis/mock/storageManagerStub.go:23]: removed

File: genesis/process/intermediate/deployLibrarySC.go [hunk 13]
```diff
@@ -26,7 +26,7 @@ type deployLibrarySC struct {
 }
 
 // NewDeployLibrarySC returns a new instance of deploy library SC able to deploy library SC that needs to be present
-// on all shards - same contract is deployed core.MaxNumShards == 256 times with addresses which end with all
+// on all shards - same contract is deployed common.MaxNumShards == 256 times with addresses which end with all
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply genesis/process/intermediate/deployLibrarySC.go:29]: yes. it is used on `deployLibrarySC.GenerateInitialPublicKeys`

File: mock/accountWrapperMock.go [hunk 14]
```diff
@@ -4,16 +4,16 @@ package mock
 import (
 	"math/big"
 
-	"github.com/ElrondNetwork/elrond-go/data"
-	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/state"
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply mock/accountWrapperMock.go:7]: added a TODO

File: mock/evictionWaitingListMock.go [hunk 15]
```diff
@@ -103,7 +103,7 @@ func (ewl *EvictionWaitingList) IsInterfaceNil() bool {
 }
 
 // ShouldKeepHash --
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply mock/evictionWaitingListMock.go:105]: done

File: node/errors.go [hunk 16]
```diff
@@ -97,8 +97,8 @@ var ErrInvalidTransactionVersion = errors.New("invalid transaction version")
 // ErrNilBootstrapComponents signals that a nil bootstrap components instance has been provided
 var ErrNilBootstrapComponents = errors.New("nil bootstrap componennts")
 
-// ErrNilCoreComponents signals that a nil core components instance has been provided
-var ErrNilCoreComponents = errors.New("nil core components")
+// ErrNilCoreComponents signals that a nil common components instance has been provided
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply node/errors.go:100]: reverted

File: node/node.go [hunk 17]
```diff
@@ -1131,7 +1131,7 @@ func (n *Node) GetHardforkTrigger() HardforkTrigger {
 	return n.hardforkTrigger
 }
 
-// GetCoreComponents returns the core components
+// GetCoreComponents returns the common components
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply node/node.go:1135]: reverted

File: node/nodeRunner.go [hunk 18]
```diff
@@ -1128,7 +1128,7 @@ func (nr *nodeRunner) CreateManagedNetworkComponents(
 	return managedNetworkComponents, nil
 }
 
-// CreateManagedCoreComponents is the managed core components factory
+// CreateManagedCoreComponents is the managed common components factory
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply node/nodeRunner.go:1132]: reverted

File: node/options.go [hunk 19]
```diff
@@ -32,7 +32,7 @@ func WithBootstrapComponents(bootstrapComponents factory.BootstrapComponentsHand
 	}
 }
 
-// WithCoreComponents sets up the Node core components
+// WithCoreComponents sets up the Node common components
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply node/options.go:35]: reverted

File: process/errors.go [hunk 20]
```diff
@@ -10,8 +10,8 @@ var ErrNilMessage = errors.New("nil message")
 // ErrNilAccountsAdapter defines the error when trying to use a nil AccountsAddapter
 var ErrNilAccountsAdapter = errors.New("nil AccountsAdapter")
 
-// ErrNilCoreComponentsHolder signals that a nil core components holder was provided
-var ErrNilCoreComponentsHolder = errors.New("nil core components holder")
+// ErrNilCoreComponentsHolder signals that a nil common components holder was provided
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply process/errors.go:13]: reverted

File: process/interceptors/factory/argInterceptedDataFactory.go [hunk 21]
```diff
@@ -1,16 +1,16 @@
...
 	"github.com/ElrondNetwork/elrond-go/crypto"
-	"github.com/ElrondNetwork/elrond-go/data/typeConverters"
-	"github.com/ElrondNetwork/elrond-go/hashing"
-	"github.com/ElrondNetwork/elrond-go/marshal"
 	"github.com/ElrondNetwork/elrond-go/process"
 	"github.com/ElrondNetwork/elrond-go/sharding"
 )
 
-// interceptedDataCoreComponentsHolder holds the core components required by the intercepted data factory
+// interceptedDataCoreComponentsHolder holds the common components required by the intercepted data factory
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply process/interceptors/factory/argInterceptedDataFactory.go:13]: reverted

File: process/interface.go [hunk 22]
```diff
@@ -1040,7 +1040,7 @@ type FallbackHeaderValidator interface {
 	IsInterfaceNil() bool
 }
 
-// CoreComponentsHolder holds the core components needed by the interceptors
+// CoreComponentsHolder holds the common components needed by the interceptors
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply process/interface.go:1044]: reverted

File: sharding/networksharding/peerShardMapper.go [hunk 23]
```diff
@@ -198,7 +198,7 @@ func (psm *PeerShardMapper) getPeerSubType(pid core.PeerID) core.P2PPeerSubType
 
 	subType, ok := subTypeObj.(core.P2PPeerSubType)
 	if !ok {
-		log.Warn("PeerShardMapper.getPeerInfoSearchingPidInFallbackCache: the contained element should have been of type core.P2PPeerSubType")
+		log.Warn("PeerShardMapper.getPeerInfoSearchingPidInFallbackCache: the contained element should have been of type common.P2PPeerSubType")
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply sharding/networksharding/peerShardMapper.go:202]: core

File: update/errors.go (hunk 1 above)
Comment (bogdan-rosianu) [reply update/errors.go:53]: reverted

File: api/transaction/routes_test.go [hunk 24]
```diff
@@ -13,15 +13,15 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	tr "github.com/ElrondNetwork/elrond-go-core/data/transaction"
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply api/transaction/routes_test.go:17]: renamed

File: cmd/termui/view/termuic/termuiRenders/widgetsRender.go [hunk 25]
```diff
@@ -163,7 +163,7 @@ func (wr *WidgetsRender) prepareInstanceInfo() {
 
 	// TODO: repair the rewards estimation or replace these 2 rows with rating details
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply cmd/termui/view/termuic/termuiRenders/widgetsRender.go:165]: added `EN-10439`

File: dataRetriever/requestHandlers/requestHandler.go [hunk 26]
```diff
@@ -22,7 +22,7 @@ var log = logger.GetOrCreate("dataretriever/requesthandlers")
 const bytesInUint32 = 4
 const timeToAccumulateTrieHashes = 100 * time.Millisecond
 
-//TODO move the keys definitions that are whitelisted in core and use them in InterceptedData implementations, Identifiers() function
+//TODO move the keys definitions that are whitelisted in common and use them in InterceptedData implementations, Identifiers() function
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply dataRetriever/requestHandlers/requestHandler.go:25]: reverted

File: factory/mock/processComponentsStub.go [hunk 27]
```diff
@@ -1,8 +1,8 @@
 package mock
 
 import (
+	"github.com/ElrondNetwork/elrond-go/common/dblookupext"
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply factory/mock/processComponentsStub.go:4]: done

File: go.mod [hunk 28]
```diff
@@ -35,16 +33,15 @@ require (
 	github.com/libp2p/go-libp2p-transport-upgrader v0.4.2
 	github.com/libp2p/go-tcp-transport v0.2.3
 	github.com/mitchellh/mapstructure v1.4.1
-	github.com/mr-tron/base58 v1.2.0
 	github.com/multiformats/go-multiaddr v0.3.3
-	github.com/pelletier/go-toml v1.9.0
+	github.com/pelletier/go-toml v1.9.3
 	github.com/pkg/errors v0.9.1
-	github.com/shirou/gopsutil v0.0.0-20190901111213-e4ec7b275ada
+	github.com/shirou/gopsutil v3.21.6+incompatible
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply go.mod:39]: tried go mod tidy and no change

File: integrationTests/state/stateTrie/stateTrie_test.go [hunk 29]
```diff
@@ -16,26 +16,26 @@ import (
...
+	"github.com/ElrondNetwork/elrond-go/state/temporary"
 	"github.com/ElrondNetwork/elrond-go/storage"
 	"github.com/ElrondNetwork/elrond-go/storage/memorydb"
 	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
 	"github.com/ElrondNetwork/elrond-go/testscommon"
+	"github.com/ElrondNetwork/elrond-go/trie"
+	factory2 "github.com/ElrondNetwork/elrond-go/trie/factory"
 	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply integrationTests/state/stateTrie/stateTrie_test.go:41]: used require

File: integrationTests/state/stateTrie/stateTrie_test.go [hunk 30]
```diff
@@ -16,26 +16,26 @@ import (
...
+	"github.com/ElrondNetwork/elrond-go/state/factory"
+	"github.com/ElrondNetwork/elrond-go/state/storagePruningManager"
+	"github.com/ElrondNetwork/elrond-go/state/storagePruningManager/evictionWaitingList"
+	"github.com/ElrondNetwork/elrond-go/state/temporary"
 	"github.com/ElrondNetwork/elrond-go/storage"
 	"github.com/ElrondNetwork/elrond-go/storage/memorydb"
 	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
 	"github.com/ElrondNetwork/elrond-go/testscommon"
+	"github.com/ElrondNetwork/elrond-go/trie"
+	factory2 "github.com/ElrondNetwork/elrond-go/trie/factory"
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply integrationTests/state/stateTrie/stateTrie_test.go:38]: done

File: integrationTests/state/stateTrie/stateTrie_test.go [hunk 31]
```diff
@@ -16,26 +16,26 @@ import (
 	"testing"
 	"time"
 
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/pubkeyConverter"
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	transaction2 "github.com/ElrondNetwork/elrond-go-core/data/transaction"
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply integrationTests/state/stateTrie/stateTrie_test.go:22]: done

File: node/blockAPI/baseBlock.go [hunk 32]
```diff
@@ -33,8 +33,8 @@ type baseAPIBockProcessor struct {
 	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter
 	historyRepo              dblookupext.HistoryRepository
 	// TODO: use an interface instead of this function
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply node/blockAPI/baseBlock.go:35]: added `EN-10440`

File: node/nodeRunner.go [hunk 33]
```diff
@@ -199,7 +199,7 @@ func (nr *nodeRunner) executeOneComponentCreationCycle(
 	flagsConfig := configs.FlagsConfig
 	configurationPaths := configs.ConfigurationPathsHolder
 
-	log.Debug("creating core components")
+	log.Debug("creating common components")
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply node/nodeRunner.go:203]: reverted

File: process/txsimulator/wrappedAccountsDB.go (hunk 2 above)
Comment (bogdan-rosianu) [reply process/txsimulator/wrappedAccountsDB.go:81]: renamed to `r`

File: storage/txstatus/txStatusComputer.go [hunk 34]
```diff
@@ -1,38 +1,18 @@
...
-)
-
-// String returns the string representation of the status
-func (tx TxStatus) String() string {
-	return string(tx)
-}
+var log = logger.GetOrCreate("storage/txstatus")
 
 // statusComputer computes a transaction status
 type statusComputer struct {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply storage/txstatus/txStatusComputer.go:18]: moved

**Review by iulianpascalau — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.
The new error will be solved with another PR. 

image

**Pull request merged**